src/libraries/System.Formats.Tar/tests/TarReader/TarReader.File.GlobalExtendedAttributes.Tests.cs

```diff
@@ -82,5 +82,27 @@ public void ExtractGlobalExtendedAttributesEntry_Throws()
                 Assert.Throws<InvalidOperationException>(() => entry.ExtractToFile(Path.Join(root.Path, "file"), overwrite: true));
             }
         }
+
+        [Theory]
+        [InlineData("key", "value")]
+        [InlineData("key     ", "   value    ")]
+        [InlineData("      key     ", "   value    ")]
+        [InlineData("many sla/s\\hes", "/////////////\\\\\\///////////")]
+        public void GlobalExtendedAttribute_Roundtrips(string key, string value)
+        {
+            var stream = new MemoryStream();
+            using (var writer = new TarWriter(stream, leaveOpen: true))
+            {
+                writer.WriteEntry(new PaxGlobalExtendedAttributesTarEntry(new Dictionary<string, string>() { { key, value } }));
+            }
+
+            stream.Position = 0;
+            using (var reader = new TarReader(stream))
+            {
+                PaxGlobalExtendedAttributesTarEntry entry = Assert.IsType<PaxGlobalExtendedAttributesTarEntry>(reader.GetNextEntry());
+                Assert.Equal(1, entry.GlobalExtendedAttributes.Count);
+                Assert.Equal(KeyValuePair.Create(key.TrimStart(), value), entry.GlobalExtendedAttributes.First());
```

Comment:
@carlossanlop, can you also speak to my comment here?

Reply:
What's `RemoveEmptyEntries`?

Reply:
I'm reading the specs to determine if we need the trimming or not.

This is the line that trims prefix whitespace:  https://github.com/dotnet/runtime/blob/060a1967649469fa71986dc92bb41629c0970946/src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs#L724

Reply:
tl;dr: The start-trimming isn't needed for the key. It could be used for the value though, but it may not be necessary.

Here's what I found:

The `pax` tool instructions indicate that extended attributes can be added with the `-o` argument, and explicitly indicates that preceding spaces are allowed:

> -o  options
> Provide information to the implementation to modify the algorithm for extracting or writing files. The value of options shall consist of one or more <comma>-separated keywords of the form:
> `keyword[[:]=value][,keyword[[:]=value], ...]`
> [...]
> **Keywords can be preceded with white space.** The value field shall consist of zero or more characters; [...]
Source: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/pax.html#tag_20_92_13_03

The `tar` tool instructions do not indicate if whitespace is allowed, but it says that it uses the same rules as pax:

> ‘--pax-option=keyword-list’
> Handle keywords in PAX extended headers. This option is equivalent to ‘-o’ option of the pax utility.
Source: https://www.gnu.org/software/tar/manual/html_node/PAX-keywords.html#index-pax_002doption

I created an archive using `tar` with a global extended attribute header. It includes one entry where both the key and the value surrounded by whitespace. The result is that the tool _only_ trims the preceding whitespace from the _value_, not from the _key_.

Commands:
```
$ ls
mount.txt
$ tar cvf destination.tar mount.txt --pax-option="  name  =  value  "
mount.txt
$ ls
mount.txt destination.tar
```

Hex Editor screenshot:
![image](https://user-images.githubusercontent.com/1175054/202299625-fe39811b-203f-4886-87d0-c44e8375c268.png)

Reply:
> What's RemoveEmptyEntries?

The code prior to my perf changes was doing:
```C#
... = line.Split(' ', 2, StringSplitOptions.RemoveEmptyEntries);
```
That has the effect of trimming all starting whitespace, because the separator specified is a space, which means each one ends up creating a new zero-length entry that gets removed due to RemoveEmptyEntries.  That's why my change added the skipping of starting whitespace, to mimic what was there before.

Reply:
> The start-trimming isn't needed for the key. 

Isn't "needed" or isn't "desired"? i.e. it's a bug that we do it and we should remove it?

> It could be used for the value though, but it may not be necessary.

You can have paths that start with spaces, just as you can have paths with interior spaces.  If we're concerned about paths that contain spaces, presumably we're also concerned about ones that start with spaces?

Reply:
@carlossanlop was there a motivation for the `TrimStart()` call? If it has been there since the beginning of S.F.Tar, maybe it was just a wrong assumption of the spec, which doesn't say anything about trimming.

If you comment-out that method, does any test break?

>  the tool only trims the preceding whitespace from the value, not from the key.

Does the same occur with BSD Tar? A potential reason to keep the trimming is to align the code with the tar tools, only on key; on value, cases like the "path starting with whitespace" that @stephentoub mentioned should be supported.

IMO: we should just remove it since is not in the spec.

Reply:
It was a wrong assumption. I also agree we should remove it since it's not in the spec.